flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -1207,7 +1207,7 @@ protected int getNumberRequiredTaskManagerSlots() {
 
 	public static Collection<ResourceProfile> createWorkerSlotProfiles(Configuration config) {
 		final int numSlots = config.getInteger(TaskManagerOptions.NUM_TASK_SLOTS);
-		final long managedMemoryBytes = MemorySize.parse(config.getString(TaskManagerOptions.LEGACY_MANAGED_MEMORY_SIZE)).getBytes();
+		final long managedMemoryBytes = MemorySize.parse(config.getString(TaskManagerOptions.MANAGED_MEMORY_SIZE, "0b")).getBytes();
```

Comment:
The reason we need this hack is because the blink operators do not support separated on-heap/off-heap managed memory atm (see `NodeResourceUtil#fromManagedMem`). If we set both memory types, then blink jobs can immediately fail under off-heap managed memory setups.

Reply:
ok, thanks for explanation. I would still try to remove `ActiveResourceManagerFactory/Test` and keep hack inside `createWorkerSlotProfiles`. E.g. set heap managed memory as a sum of heap/off-heap (with a detailed explaining comment referencing the follow-up issue) but from `TaskExecutorResourceSpec`, not by adjusted config in `ActiveResourceManagerFactory`, this looks confusing, IMO.